subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -445,16 +444,46 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 		return;
 	}
 
-	prov_clear_tx();
-
-	link.rx.seg = (1 << (START_LAST_SEG(rx->gpc) + 1)) - 1;
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
+
+	if (link.rx.seg & BIT(0)) {
+		seg = ~link.rx.seg;
+		BT_WARN("Seg Start package losted, (seg: 0x%02x)", seg);
+
+		seg = find_msb_set(seg) - 1;
+		if (seg > link.rx.last_seg) {
+			BT_ERR("Invalid segment index %u", seg);
+			prov_failed(PROV_ERR_NVAL_FMT);
+			return;
+		}
+	}
+
+	if (link.rx.seg) {
+		seg = link.rx.seg;
+	}
+
+	link.rx.seg = (seg & (1 << (START_LAST_SEG(rx->gpc) + 1))) - 1;
 	memcpy(link.rx.buf->data, buf->data, buf->len);
 	XACT_SEG_RECV(0);
 
-	if (!link.rx.seg) {
-		prov_msg_recv();
+	if (link.rx.seg) {
+		return;
 	}
+
+	if (link.rx.last_seg) {
+		uint8_t expect_len;
+
+		expect_len = (link.rx.buf->len - 20U -
+			      ((link.rx.last_seg - 1) * 23U));
+		if (expect_len != link.rx.buf->len) {
```

Comment:
This check will always fail. 

If we receive the last segment before the start segment, there's no way to perform this check with the current design. The `link.rx.buf->len` would need a different approach to work for all scenarios.